src/utils/formattedTime.ts

```diff
@@ -153,11 +155,30 @@ function getRelativeDay(
 	return idiomaticTimeFormatter.format(diffInDays, 'day')
 }
 
+/**
+ * Generates duration string given days, hours, minutes, seconds
+ * e.g. '1 day', '2 hours', '3 minutes'
+ *
+ * @param data
+ * @param data.days number of days
+ * @param data.hours number of hours
+ * @param data.minutes number of minutes
+ * @param data.seconds number of seconds
+ */
+function generateDurationString({ days, hours, minutes, seconds } = { days: 0, hours: 0, minutes: 0, seconds: 0 }) {
+	if (!durationFormatter) {
+		return null
+	}
+
+	return durationFormatter.format({ days, hours, minutes, seconds })
+}
```

Comment:
I don't think we need a fallback values here, it works just fine with undefined. And with declared types, I'd pass just this:

```suggestion
function generateDurationString((duration: DurationInput)) {
	return durationFormatter.format(duration)
}
```